util/src/main/scala/scala/scalanative/io/VirtualDirectory.scala

```diff
@@ -86,11 +92,36 @@ object VirtualDirectory {
       } finally stream.close()
     }
 
+    override def write(path: Path, fn: Writer => Unit): Path = {
+      val fullPath = resolve(path)
+      val writer   = Files.newBufferedWriter(fullPath)
+      try fn(writer)
+      finally writer.close()
+      fullPath
+    }
+
     override def write(path: Path, buffer: ByteBuffer): Unit = {
       val channel = open(resolve(path))
       try channel.write(buffer)
       finally channel.close
     }
+
+    override def merge(sources: Seq[Path], target: Path): Unit = {
+      if (sources.nonEmpty) {
+        val output = FileChannel.open(resolve(target),
+                                      StandardOpenOption.CREATE,
+                                      StandardOpenOption.WRITE,
+                                      StandardOpenOption.APPEND)
+        try sources.foreach { path =>
```

Comment:
This is weird formatting, IMO. Consider:
```suggestion
        try {
          sources.foreach { path =>
```